qiskit/backends/ibmq/ibmqjob.py

```diff
@@ -435,13 +440,19 @@ def _wait_for_submission(self, timeout=60):
                 raise JobError(str(submit_info['error']))
 
 
-def _reorder_bits(result):
+def _reorder_bits(job_data):
     """Temporary fix for ibmq backends.
 
     For every ran circuit, get reordering information from qobj
     and apply reordering on result.
+
+    Args:
+        job_data (dict): dict with the bare contents of the API.get_job request.
+
+    Raises:
+            JobError: raised if the creg sizes don't add up in result header.
     """
-    for circuit_result in result._result['result']:
+    for circuit_result in job_data['qasms']:
```

Comment:
This assumes that the content returned from the API will have a `qasms` field, which is not true for the new format (see #713). Let's merge that PR first, then make sure that qiskit handles things well in both old and new format.

Reply:
... or actually we can just merge this, and then in #713 do the necessary final adjustments for the new/old format coming from backends.

Just please fix the `test_reordering` test, and we should be good to go.

Reply:
@delapuente hmm.. my comment got lost. I had commented that in order to fix the `test_reordering:test_multi_register_reordering` test, you just need to replace:
```
 circuit.ccx(qr2[0], qr1[1], qr1[0])
```
with
```
circuit.cx(qr2[0], qr1[0])
```

Reply:
Test [`test_multi_register_reordering`](https://github.com/Qiskit/qiskit-terra/blob/master/test/python/test_reordering.py#L56) is intermittent and has been failing since, at least, to 0.5.7. I've open an issue to keep track of it (@848) but, since the PR does not introduce a regression, I'm merging now.

Reply:
I miss your comment and open #848. Merging this and fixing there. I've tested the failing is not a regressing coming from this bug since the test has been failing since `0.5.7` or before.